Update user-interface-text-guidelines.md (#836)
mattermost/mattermost-handbook: operations/research-and-development/product/development-process/user-interface-text-guidelines.md

# User Interface Text Guidelines

Mattermost user interface content (microcopy) is where we can set our users up for success and build engagement with the product instead of reliance on documentation.

When we write copy for our product messaging our primary goals are to:

* Write with empathy
* Provide context
* Help users build mental models
* Reduce cognitive load
* Proactively prevent failure

To achieve these goals, we ensure that we have the right content for the right user at the right time. In addition to this, we prioritize:

> * Plain language: We use natural language for documentation and in-product text.

Reviewer: ```suggestion
* Plain language: We choose natural language, simple words, and short sentences for documentation and in-product text.
```

* Readability: We strive to make our documentation pages easy to scan and not overwhelming.
* Consistency: We ensure that we all refer to the same things using the same words or names, to avoid confusion.

> These writing principles mean that we think like a friend, and talk like a coach to reduce friction and increase satisfaction. 

Reviewer: I love this. Great description!

* Keep in mind that your users are from all over the world. 
* Use the present tense to describe a current state or condition, and the future tense to state something that is very definitely going to happen.
* Use the active voice, except for these cases:
  * If you’ll end up blaming the user. For example, don’t say **You entered an incorrect password**. Instead, say **The password is incorrect**.
  * If you’re describing what just happened. For example: **Your incoming webhook is set up**.
  * If the subject \(the doer of an action\) is the Mattermost application itself. For example: **The image has been deleted** instead of **The server deleted the image**.

> * If you’re asking the user to do or not do something, use imperatives \(command phrases\). For example, use **Don't change the Hostname** instead of **It's not recommended to change the Hostname**. Better still, explain what could go wrong if they do or don’t do something. For example: **Don't change the Hostname because doing so**.

Reviewer: ```suggestion
* If you’re asking the user to do or not do something, use imperatives \(command phrases\). For example, use **Don't change the Hostname** instead of **It's not recommended to change the Hostname**. Better still, explain what could go wrong if they do or don’t do something. For example: **Don't change the Hostname because doing so will break something**.
```

## Types of in-product content

In general, users will encounter the following types of messaging when using Mattermost:

> * Help/info text

Reviewer: ```suggestion
* Help, hints, or informative text
```

> * Calls to Action

Reviewer: ```suggestion
* Calls to action
```

* Confirmation dialogs
* Errors
* Notifications

In general, this content is either:

> * Triggered because something happened

Reviewer: Consider reordering these 3 items to match the content above (to introduce more parallelism):
- static text
- actionable button
- triggered

* If a solution or workaround isn’t obvious, suggest one.
* Examples:
  * **Messages must have fewer than 120 characters.**
  * **A connection to the Marketplace server could not be established. Check your settings in the **System Console**.**

## In-product copy

The Technical Writers often work with the UI/UX team on in-product copy. The majority of guidelines are available in the [Mattermost Compass Design System](https://zeroheight.com/29be2c109/p/995619-compass-design-system).

> ### Guidelines for punctuation

Author: I'm not entirely sure we need this section - might be better to replace it with a sentence instead?

Reviewer: I wonder if a table would work well here? 4 columns: guideline, description, good example, bad example

Author: Oh yes, that's a great idea! Will add that into the next iteration, thanks!

**Headings**

These are H1 headings such as the title of a modal. Titles shouldn't have periods unless the headline is more than one sentence. Titles also shouldn't contain punctuation such as question marks, colons, semi-colons.

**Paragraphs**

Paragraphs should always be properly punctuated.

**Bullets**

Bullet lists shouldn't have periods unless the bullet text is more than one sentence.

**Button labels**

Button labels shouldn't have periods or other punctuation.

### Guidelines for capitalization

We follow the same capitalization rules across all our documentation assets, including in-product text and UI elements: Always use sentence case (except where you're using a proper noun, of course). This applies to:

* Page titles
* Page headings
* Section headings
* Button labels
* Input labels
* Navigation labels

> * Menu items

Reviewer: + field hint text & hover text

**Instead of:** This Article is About Mattermost
**Use:** This article is about Mattermost

### How we write things (WIP)

> As previously mentioned, we write with empathy in the context of achievement, and in natural language. This can be quite hard to do; as technical writers we generally lean toward very clear and concise writing that can feel clinical. Writing more naturally means we try to avoid convoluted phrasing and we try to make things more simple. 

Reviewer: ```suggestion
As previously mentioned, we write with empathy in the context of achievement using natural language. This can be quite hard to do; as technical writers we generally lean toward very clear and concise writing that can feel clinical. Writing more naturally means we try to avoid convoluted phrasing and we try to make things more simple. 
```

So, instead of saying: "When testing the Java app ensure your third-party connection is secure before initiating the test sequence." rather say: "Make sure your third-party connection is secure before testing the app."

Because we strive to use natural language, the way we phrase microcopy isn't templated - that would defeat the point. Instead, here are some examples of microcopy we use for various features and products. The common theme is clarity and empathy.

> 

Author: I'll add some screenshots of the playbooks microcopy here.

Reviewer: Great idea!
